src/hotspot/share/opto/cfgnode.cpp

```diff
@@ -1779,6 +1779,66 @@ static Node* is_absolute( PhaseGVN *phase, PhiNode *phi_root, int true_path) {
   return x;
 }
 
+static Node* is_minmax(PhaseGVN* phase, PhiNode* phi_root, int true_path) {
+  assert(true_path != 0, "only diamond shape graph expected");
+
+  Node* region = phi_root->in(0);
+  IfNode* iff = region->in(1)->in(0)->as_If();
+  BoolNode* bol = iff->in(1)->as_Bool();
+  Node* cmp = bol->in(1);
+
+  // Don't transform if the branch is highly predictable
+  constexpr float infrequent_prob = PROB_UNLIKELY_MAG(2);
+  if (iff->_prob < infrequent_prob || iff->_prob > (1.0f - infrequent_prob)) {
+    return nullptr;
+  }
+
+  int opcode = cmp->Opcode();
+
+  // Ensure comparison is an integral type
+  if (opcode != Op_CmpI && opcode != Op_CmpL) {
+    return nullptr;
+  }
+
+  // Only accept canonicalized le and lt comparisons
+  int test = bol->_test._test;
+  if (test != BoolTest::le && test != BoolTest::lt) {
+    return nullptr;
+  }
+
+  int false_path = 3 - true_path;
+
+  // The values being compared
+  Node* cmp_true = cmp->in(true_path);
+  Node* cmp_false = cmp->in(false_path);
+
+  // The values being selected
+  Node* phi_true = phi_root->in(true_path);
+  Node* phi_false = phi_root->in(false_path);
+
+  // For this transformation to be valid, the values being compared must be the same as the values being selected.
+  // We accept two different forms, "a < b ? a : b", and "a < b ? b : a". For the first form, the comparison is a minimum
+  // if 'a' (true_path == 1) is selected when the comparison is true. If 'b' (true_path == 2) is selected when it is true,
+  // the comparison will be a maximum. For the second form, as the order of the selected values is swapped, the selected
+  // paths must also be swapped, so the path to choose maximum is 2 instead of 1. If neither form is found, bail out.
+
+  int max_path;
```

Comment:
As a heads up, in this latest change I've changed the code from identifying the minimum path to identifying the maximum path, to fit better with `MaxNode::build_min_max_int` taking in an `is_max` argument.